tools/nightly-unit-tests/src/bitrise_yml_writter.py

```diff
@@ -0,0 +1,74 @@
+# -----------------------------------------------------------
+# Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019-2020 Datadog, Inc.
+# -----------------------------------------------------------
+
+from src.semver import Version
+from dataclasses import dataclass
+
+
+@dataclass()
+class UnitTestsWorkflow:
+    simulator_device_name: str
+    simulator_os_name: str
+    simulator_os_version: Version
+
+    def get_invocation_yml(self):
+        return f'    - _run_unit_tests_on_{self.simulator_os_version}'
+
+    def get_definition_yml(self):
+        return f'''
+  _run_unit_tests_on_{self.simulator_os_version}:
+    envs:
+    - SIMULATOR_DEVICE: '{self.simulator_device_name}'
+    - SIMULATOR_OS_VERSION: '{self.simulator_os_version}'
+    after_run:
+    - _run_unit_tests
+'''[1:-1]
+
+    def get_slack_report_row(self):
+        return f'            - tested on {self.simulator_os_name} {self.simulator_os_version} - {self.simulator_device_name}'
```

Comment:
why do we have whitespaces here and in `get_invocation_yml` above?
if it's for indentation, maybe we can do something like `f'{indent(level=2)}-tested on ...`  by adding a `def indent(level): ...`?

Reply:
This is to ensure YML syntax correctness, e.g. workflow invocations should be properly intended by 4 spaces:
```yml
workflows:
  run_all:
    after_run:
    - _make_dependencies
### <WORKFLOW INVOCATIONS> ###
    - _send_slack_notification
```

I addressed this feedback, but instead of hardcoding `level` variable (which is similar to hardcoding space characters), I made the variables rendering dynamic. Now the variable is intended by the same number of whitespaces as it was defined with:
```yml
workflows:
  run_all:
    after_run:
    - _make_dependencies
    ### <WORKFLOW INVOCATIONS> ###
    - _send_slack_notification
```
```python
def get_invocation_yml(self):
    return f'- _run_unit_tests_on_{self.simulator_os_version}'
```